modules/javafx.graphics/src/main/native-glass/monocle/epd/mxcfb.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2004-2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ */
+
+/*
+ * The code contained herein is licensed under the GNU Lesser General
+ * Public License.  You may obtain a copy of the GNU Lesser General
+ * Public License Version 2.1 or later at the following locations:
+ *
```

Comment:
This is third-party code, so we cannot accept this file.

Reply:
Thank you, Kevin. I considered the license compatibility but forgot about the Oracle Contributor Agreement. If I understand the history correctly, David Hill ran into the same problem back in 2013. Look for `mxcfb.h` in the bug reports [JDK-8103599](https://bugs.openjdk.java.net/browse/JDK-8103599 "Add support for imx6") and [JDK-8103668](https://bugs.openjdk.java.net/browse/JDK-8103668 "Add support for imx6 - modify gradle files to fetch armv6-vfp-04.tgz"). There are two solutions to the problem still in the code today.

The first solution was to copy the header file manually to the directory for the ARM Linux system headers. The script that sets up the cross-compile toolchain still reminds you to do so:

```ShellSession
$ ../buildSrc/crosslibs/crosslibs-armv6hf.sh
...
WARNING: remember to hand patch in usr/include/linux/mxcfb.h for i.mx cursor support

Done.
```

The header file is never used, though, because a better solution hard-coded the field offsets in their C structures. For example, see the nested classes `MXCFBColorKey`, `MXCFBGblAlpha`, and `MXCFBPos` in the file [MX6Cursor.java](https://github.com/javafxports/openjdk-jfx/blob/develop/modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/MX6Cursor.java). I should be able to use the same technique in my code.

There is another third-party file in my pull request. The file is [16777216colors diffpatt.png](https://commons.wikimedia.org/wiki/File:16777216colors_diffpatt.png), published under the Creative Commons "CC BY-SA 4.0" license and loaded by the unit test case [FramebufferY8Test.java](https://github.com/javafxports/openjdk-jfx/blob/b3963fddc47c75b9179604562bc4c97199bacd72/modules/javafx.graphics/src/test/java/test/com/sun/glass/ui/monocle/FramebufferY8Test.java). If this file cannot be included, I could generate the image within the test case itself. Generating the image will also prevent the problem that Johan encountered.

Unless I hear otherwise, I will test both changes over the next week:

* hard-code the field offsets and delete the header file, and
* generate the image in the test case and delete the image file.

Reply:
This seems like a good solution.

Reply:
This turned out to be a good change. Removing the header file allowed me to:

* remove 208 lines of native C source code plus the 120 lines of the C header file,
* move the initialization of the IOCTL request codes into the EPDSystem constructor and make their fields final,
* compute the size of the C structures at compile time instead of at runtime through a native call, and
* eliminate all native calls when getting and setting fields in the C structures.

The removal of the header file should also make it easier to accomodate changes to the structures in later versions of the EPD Controller API.